docs/development/pr-review.asciidoc

```diff
@@ -0,0 +1,129 @@
+--
+
+
+[float]
+=== Product level review
+
+Reviewers are not simply evaluating the code itself, they are also evaluating the quality of the user-facing change in the product. This generally means they need to check out the branch locally and "play around" with it. In addition to the "do we want this change in the product" details from above, the reviewer should be looking for bugs and evaluating how approachable and useful the feature is as implemented. Special attention should be given to error scenarios and edge cases to ensure they are all handled well within the product.
+
+
+[float]
+=== Consistency, style, readability
+
+Having a relatively consistent codebase is an important part of us building a sustainable project. If 40+ different developers all wrote code in whatever style they personally desired, the resulting inconsistencies would make working across the codebase more challenging. We know this because it's happened before in the Kibana project with far fewer developers than we have contributing to the project today, and the process of retroactively getting the codebase into a consistent state was extremely time intensive and disruptive to all contributors.
+
+On the other hand, improvements are being added to JavaScript, TypeScript, and our dependencies all the time, so we should be flexible about evolving our style and coding standards over time. Absolute consistency across the whole codebase is not our goal here - instead, we seek to move together with consensus as we evolve our codebase over time.
+
+As always, automation is the best way we can help ensure consistency, so we enforce a pretty comprehensive set of linting rules through CI. We're also rolling out prettier to make this even more automatic.
+
+For things that can't practically get caught by a linter, we maintain various styleguides that authors should adhere to and reviewers should keep in mind when they review a pull request.
+
+Beyond that, we're into subjective territory. Statements like "this isn't very readable" are hardly helpful since they can't be qualified, but that doesn't mean a reviewer should outright ignore code that is hard to understand due to how it is written. There isn't one definitively "best" way to write any particular code, so pursuing such shouldn't be our goal. Instead, reviewers and authors alike must accept that there are likely many different appropriate ways to accomplish the same thing with code, and so long as the contribution is utilizing one of those ways, then we're in good shape.
+
+On the other hand, if a reviewer calls out a specific chunk of code as being hard to understand, then one of the first people to look at the code is having a hard time understanding it. If the reviewer feels strongly that this is a problem, they should propose alternatives for the author's consideration.
```

Comment:
I think it should be okey to say something along the lines of: "I'm struggling to follow the flow in this function. Got any ideas for how we can make it clearer?" One of the problems I've had with proposing alternatives is that I end up pulling the code and re-working it locally and including a patch in the PR. It feels a bit "heavy" at times, but maybe it's just okey.

Reply:
I'll reword this. The key thing I'm trying to underscore here is that in these situations the reviewer needs to be pragmatic and propel this forward. We shouldn't block PRs are on entirely nebulous things.

Reply:
Reworded this